superset-frontend/src/SqlLab/components/ScheduleQueryButton.tsx

```diff
@@ -0,0 +1,208 @@
+import { t, styled } from '@superset-ui/core';
+import chrono from 'chrono-node';
+import ModalTrigger from 'src/components/ModalTrigger';
+import FormLabel from 'src/components/FormLabel';
+import './ScheduleQueryButton.less';
+import Button from 'src/components/Button';
+
+const validators = {
+  greater: (a: number, b: number) => a > b,
+  greater_equal: (a: number, b: number) => a >= b,
+  less: (a: number, b: number) => a < b,
+  less_equal: (a: number, b: number) => a <= b,
+};
+
+const getJSONSchema = () => {
+  const jsonSchema = window.featureFlags.SCHEDULED_QUERIES?.JSONSCHEMA;
+  // parse date-time into usable value (eg, 'today' => `new Date()`)
+  if (jsonSchema) {
+    Object.entries(jsonSchema.properties).forEach(
+      ([key, value]: [string, any]) => {
+        if (value.default && value.format === 'date-time') {
+          jsonSchema.properties[key] = {
+            ...value,
+            default: chrono.parseDate(value.default).toISOString(),
+          };
+        }
+      },
+    );
+    return jsonSchema;
+  }
+  return {};
+};
+
+const getUISchema = () => window.featureFlags.SCHEDULED_QUERIES?.UISCHEMA;
+
+const getValidationRules = () =>
+  window.featureFlags.SCHEDULED_QUERIES?.VALIDATION || [];
+
+const getValidator = () => {
+  const rules: any = getValidationRules();
+  return (formData: Record<string, any>, errors: FormValidation) => {
+    rules.forEach((rule: any) => {
+      const test = validators[rule.name];
+      const args = rule.arguments.map((name: string) => formData[name]);
+      const container = rule.container || rule.arguments.slice(-1)[0];
+      if (!test(...args)) {
+        errors[container].addError(rule.message);
+      }
+    });
+    return errors;
+  };
+};
+
+interface ScheduleQueryButtonProps {
+  defaultLabel?: string;
+  sql: string;
+  schema?: string;
+  dbId: number;
+  animation?: boolean;
+  onSchedule?: Function;
+  scheduleQueryWarning: string | null;
+  disabled: boolean;
+  tooltip: string;
+}
+
+const StyledRow = styled(Row)`
+  padding-bottom: ${({ theme }) => theme.gridUnit * 2}px;
+`;
+
+const ButtonComponent = styled.div`
+  background-color: none;
+  .ant-btn.superset-button {
+    color: rgba(0, 0, 0, 0.85);
+    text-transform: none;
+    padding-right: 0px;
+    padding: 0px;
+    font-size: 14px;
+    height: 22px;
+    font-weight: ${({ theme }) => theme.typography.weights.normal};
+    &:first-child {
+      background: none;
+    }
+  }
+`;
+
+const ScheduleQueryButton: FunctionComponent<ScheduleQueryButtonProps> = ({
+  defaultLabel = t('Undefined'),
+  sql,
+  schema,
+  dbId,
+  onSchedule = () => {},
+  scheduleQueryWarning = null,
+  tooltip = '',
+}) => {
+  const [description, setDescription] = useState('');
+  const [label, setLabel] = useState(defaultLabel);
+  const [showSchedule, setShowSchedule] = useState(false);
+  let saveModal: any;
+
+  const onScheduleSubmit = ({ formData }: Record<string, any>) => {
```

Comment:
does this formData have a type? Can you do something like `({ formData }: {formData: HTMLFormData})`? That's just a guess on the type

Reply:
is this a second deconstruction? {formData: HTMLFormData}